src/Examine/SearchResultsBase.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace Examine
+{
+    public abstract class SearchResultsBase : ISearchResults
+    {
+        public SearchResultsBase()
+        {
+            Results = new Dictionary<int, ISearchResult>();
+        }
+
+        ///<summary>
+        /// Returns an empty search result
+        ///</summary>
+        ///<returns></returns>
+        public static ISearchResults Empty()
+        {
+            return EmptySearchResults.Instance;
+        }
+
+        /// <summary>
+        /// Results for the search
+        /// </summary>
+        protected IDictionary<int, ISearchResult> Results { get; }
+
+        /// <summary>
+        /// Gets the total number of results for the search
+        /// </summary>
+        /// <value>The total items from the search.</value>
+        public long TotalItemCount { get; internal set; }
+
+        /// <summary>
+        /// Gets the total number of documents while enumerating results
+        /// </summary>
+        //NOTE: This is totally retarded but it is required for medium trust as I cannot put this code inside the Skip method... wtf
```

Comment:
we can remove this comment now we don't have med trust anymore :P

Reply:
Ha cool, consider it removed!